### ISSUE TITLE
really can't work with my HP04

### ISSUE THREAD

**Issue by saiping** [joe-ng/homebridge-dyson-link#82]
[4/23/2019, 7:50:42 PM] [DysonPlatform] Could not login to Dyson
[4/23/2019, 7:50:42 PM] [DysonPlatform] Missing password for device with serial number ZY9-CN-XXXXXXXX, devices found on your account:

the log were like this
and my json file is this:
 {
            "platform": "DysonPlatform",
            "name": "DysonPlatform",
            "email": "XXXXXhuang@gmail.com",
            "password": "xxxxxxx",
            "country": "cn",
            "accessories": [
                {
                    "ip": "192.168.1.6",
                    "displayName": "dyson",
                    "serialNumber": "ZY9-CN-xxxxxxxx"
                }
            ]
        },

and whatever i try to figure out,and just can't do it.
please help...it takes me five night for this.
can't sleep..orz

**Comment by octaviz** [comment]
If it helps you for something this is my code in my HB, it works for me

`{
            "platform": "DysonPlatform",
            "name": "DysonPlatform",
            "email": "xxxxxx@gmail.com",
            "password": "xxxxxxxxxx",
            "country": "ES",
            "accessories": [
                {
                    "ip": "192.168.3.85",
                    "displayName": "Purificador",
                    "serialNumber": "XXX-XX-XXXXXXX"
                }
            ]
        },`

**Comment by saiping** [comment]
> If it helps you for something this is my code in my HB, it works for me
> 
> `{ "platform": "DysonPlatform", "name": "DysonPlatform", "email": "xxxxxx@gmail.com", "password": "xxxxxxxxxx", "country": "ES", "accessories": [ { "ip": "192.168.3.85", "displayName": "Purificador", "serialNumber": "XXX-XX-XXXXXXX" } ] },`

thx for your help
but still no go.
I'm guessing it's cn account can't link.
my hp04 is bought from China,they didn't put this device on the sever,but the dyson link is worked.
really weird!

**Comment by joe-ng** [comment]
yup, it seemed that the Dyson API URL would be different for China devices.
I just released a new version that would connect to China server if your country code is CN.
you may try that to see if that works for you

**Comment by saiping** [comment]
> yup, it seemed that the Dyson API URL would be different for China devices.
> I just released a new version that would connect to China server if your country code is CN.
> you may try that to see if that works for you

[5/1/2019, 5:05:08 PM] [DysonPlatform] Could not login to Dyson
[5/1/2019, 5:05:08 PM] [DysonPlatform] Missing password for device with serial number ZY9-CN-XXXXXXXX, devices found on your account: 

thx,joe~
I try replace the new version file in/homebridge,and restart docker,but still got the same error message.
thx again..maybe it's unknown problem.

**Comment by jptoucas** [comment]
I also have a HP04 bought in France, there is my config : 
 {
            "platform": "DysonPlatform",
            "name": "DysonPlatform",
            "email": "****@me.com",
            "password": "*****",
            "country": "FR",
            "accessories": [
                {
                    "ip": "192.168.1.74",
                    "displayName": "Ventilateur",
                    "serialNumber": "A1A-EU-********",
                    "password": "*****",
                    "nightModeVisible": false,
                    "focusModeVisible": false,
                    "autoModeVisible": true
                }
            ]
        }

Hope this helps

On my side I have only issue : when the fan is heater mode the home app says that and add the temperature; when the fan is refreshing, the home app says “null”.
-> I need to use the Dyson Link app to get back the refreshing mode

**Comment by ellenich** [comment]
Having trouble with our HP04 too- in the US.

```
        {
            "platform": "DysonPlatform",
            "name": "DysonPlatform",
            "email": "********@me.com",
            "password": "**********",
            "country": "US",
            "accessories": [
                {
                    "ip": "192.168.*.***",
                    "displayName": "Dyson Fan",
                    "serialNumber": "**********",
                    "password": "***********"
                }
            ]
        } 
```
Kind of new to HomeBridge, but hopefully this helps. The logs look normal and I see the various features in the Home app, but they hang on "Updating" and then eventually go to "No Response".

Logs-

`
[5/11/2019, 9:47:50 PM] [DysonPlatform] Finished launching. Start to create accessory from config
[5/11/2019, 9:47:50 PM] Homebridge is running on port 51826.
[5/11/2019, 9:47:53 PM] [DysonPlatform] Use device password from account
[5/11/2019, 9:47:53 PM] [DysonPlatform] Dyson Fan IP:192.168.*.*** Serial Number:DYSON-********
[5/11/2019, 9:47:53 PM] [DysonPlatform] Device serial number format valids
[5/11/2019, 9:47:53 PM] [DysonPlatform] Device cached. Try to update this
[5/11/2019, 9:47:53 PM] [DysonPlatform] Init Sensor for Dyson Fan
[5/11/2019, 9:47:53 PM] [DysonPlatform] Init Fan State for Dyson Fan
[5/11/2019, 9:47:53 PM] [DysonPlatform] Auto switch found. Remove this now
[5/11/2019, 9:47:53 PM] [DysonPlatform] Night mode button is added
[5/11/2019, 9:47:53 PM] [DysonPlatform] Heat Available. Add Heat button and jet control
HAP Warning: Characteristic 000000BF-0000-1000-8000-0026BB765291 not in required or optional characteristics for service 000000B7-0000-1000-8000-0026BB765291. Adding anyway.
[5/11/2019, 9:47:53 PM] [DysonPlatform] Jet Focus mode button is added
`

**Comment by saiping** [comment]
STILL NOT WORKING,after the dyson link app just updated

**Comment by alexlijw** [comment]
> Having trouble with our HP04 too- in the US.
> 
> ```
>         {
>             "platform": "DysonPlatform",
>             "name": "DysonPlatform",
>             "email": "********@me.com",
>             "password": "**********",
>             "country": "US",
>             "accessories": [
>                 {
>                     "ip": "192.168.*.***",
>                     "displayName": "Dyson Fan",
>                     "serialNumber": "**********",
>                     "password": "***********"
>                 }
>             ]
>         } 
> ```
> 
> Kind of new to HomeBridge, but hopefully this helps. The logs look normal and I see the various features in the Home app, but they hang on "Updating" and then eventually go to "No Response".
> 
> Logs-
> 
> `[5/11/2019, 9:47:50 PM] [DysonPlatform] Finished launching. Start to create accessory from config [5/11/2019, 9:47:50 PM] Homebridge is running on port 51826. [5/11/2019, 9:47:53 PM] [DysonPlatform] Use device password from account [5/11/2019, 9:47:53 PM] [DysonPlatform] Dyson Fan IP:192.168.*.*** Serial Number:DYSON-******** [5/11/2019, 9:47:53 PM] [DysonPlatform] Device serial number format valids [5/11/2019, 9:47:53 PM] [DysonPlatform] Device cached. Try to update this [5/11/2019, 9:47:53 PM] [DysonPlatform] Init Sensor for Dyson Fan [5/11/2019, 9:47:53 PM] [DysonPlatform] Init Fan State for Dyson Fan [5/11/2019, 9:47:53 PM] [DysonPlatform] Auto switch found. Remove this now [5/11/2019, 9:47:53 PM] [DysonPlatform] Night mode button is added [5/11/2019, 9:47:53 PM] [DysonPlatform] Heat Available. Add Heat button and jet control HAP Warning: Characteristic 000000BF-0000-1000-8000-0026BB765291 not in required or optional characteristics for service 000000B7-0000-1000-8000-0026BB765291. Adding anyway. [5/11/2019, 9:47:53 PM] [DysonPlatform] Jet Focus mode button is added`


-----------------------------------------------------------------------------
I got the same issue with my HP02, no response if you have this plugin active. In the protocol, there is one line very suspicious:

[DysonPlatform] Heat Available. Add Heat button and jet control HAP Warning: Characteristic 000000BF-0000-1000-8000-0026BB765291 not in required or optional characteristics for service 000000B7-0000-1000-8000-0026BB765291. Adding anyway.

Does anybody know what is the issue here?

**Comment by ddorato** [comment]
Same here with a HP02, but my HP04 works fine.

**Comment by cianiandreadev** [comment]
Also for me not working with PureCool 🙁
